avalon/maya/pipeline.py

```diff
@@ -479,3 +492,8 @@ def _before_scene_save(return_code, client_data):
     OpenMaya.MScriptUtil.setBool(return_code, True)
 
     api.emit("before_save", [return_code, client_data])
+
+
+def _on_current_context_updated(*args):
+    logger.info("Running _on_current_context_updated()")
+    _set_project()
```

Comment:
In order to protect against switching to a task with a non-existent project, we should check to see whether the project exists before switching to it.

Down the line, we can introduce a dialog to the user, asking him to create the project if it doesn't exist. For this to happen, we need to separate the folder creation aspect of the Launcher into core. One step at a time.